### PR TITLE
Generate from add-v5-cluster-labeling

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#1693

### CODE REVIEW

File: @types/api/ClusterLabelsApi.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+export default class ClusterLabelsApi {
+    /**
+    * Constructs a new ClusterLabelsApi.
+    * @alias module:api/ClusterLabelsApi
+    * @class
+    * @param {module:ApiClient} apiClient Optional API client implementation to use,
+    * default to {@link module:ApiClient#instance} if unspecified.
+    */
+    constructor(apiClient: any);
+    apiClient: any;
```
Comment (axbarsan) [inline @types/api/ClusterLabelsApi.d.ts:15]:
```suggestion
    constructor(apiClient?: ApiClient);
    apiClient: ApiClient;
```

File: @types/api/ClusterLabelsApi.d.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Get cluster labels
+     * Returns the labels that this cluster is labelled with
+     * @param {String} clusterId Cluster ID
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/V5ClusterLabelsResponse} and HTTP response
+     */
+    getClusterLabelsWithHttpInfo(clusterId: string): Promise<any>;
```
Comment (axbarsan) [inline @types/api/ClusterLabelsApi.d.ts:22]:
```suggestion
    getClusterLabelsWithHttpInfo(clusterId: string): Promise<V5ClusterLabelsResponse>;
```

File: @types/api/ClusterLabelsApi.d.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Get cluster labels
+     * Returns the labels that this cluster is labelled with
+     * @param {String} clusterId Cluster ID
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/V5ClusterLabelsResponse}
+     */
+    getClusterLabels(clusterId: string): Promise<any>;
```
Comment (axbarsan) [inline @types/api/ClusterLabelsApi.d.ts:29]:
```suggestion
    getClusterLabels(clusterId: string): Promise<V5ClusterLabelsResponse>;
```

File: @types/api/ClusterLabelsApi.d.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Set cluster labels
+     * Set cluster labels. Keys and labels should adhere to [Kubernetes labels syntax and character set](https://v1-16.docs.kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set).  Label changes should be written as a [JSON Merge Patch, RFC 7386](https://tools.ietf.org/html/rfc7386).  Changes to labels containing &#x60;giantswarm.io&#x60; is forbidden, changes to label &#x60;release.giantswarm.io/version&#x60; will be validated against available Giant Swarm releases.
+     * @param {String} clusterId Cluster ID
+     * @param {module:model/V5ClusterLabelsResponse} body Labels to attach to this cluster.
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/V5ClusterLabelsResponse} and HTTP response
+     */
+    setClusterLabelsWithHttpInfo(clusterId: string, body: any): Promise<any>;
```
Comment (axbarsan) [inline @types/api/ClusterLabelsApi.d.ts:37]:
```suggestion
    setClusterLabelsWithHttpInfo(clusterId: string, body: V5ClusterLabelsResponse): Promise<V5ClusterLabelsResponse>;
```

File: @types/api/ClusterLabelsApi.d.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * Set cluster labels
+     * Set cluster labels. Keys and labels should adhere to [Kubernetes labels syntax and character set](https://v1-16.docs.kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set).  Label changes should be written as a [JSON Merge Patch, RFC 7386](https://tools.ietf.org/html/rfc7386).  Changes to labels containing &#x60;giantswarm.io&#x60; is forbidden, changes to label &#x60;release.giantswarm.io/version&#x60; will be validated against available Giant Swarm releases.
+     * @param {String} clusterId Cluster ID
+     * @param {module:model/V5ClusterLabelsResponse} body Labels to attach to this cluster.
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/V5ClusterLabelsResponse}
+     */
+    setClusterLabels(clusterId: string, body: any): Promise<any>;
```
Comment (axbarsan) [inline @types/api/ClusterLabelsApi.d.ts:45]:
```suggestion
    setClusterLabels(clusterId: string, body: V5ClusterLabelsResponse): Promise<V5ClusterLabelsResponse>;
```

File: @types/model/V5ClusterLabels.d.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+*/
+export default class V5ClusterLabels {
+    /**
+    * Constructs a <code>V5ClusterLabels</code> from a plain JavaScript object, optionally creating a new instance.
+    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
+    * @param {Object} data The plain JavaScript object bearing properties of interest.
+    * @param {module:model/V5ClusterLabels} obj Optional instance to populate.
+    * @return {module:model/V5ClusterLabels} The populated <code>V5ClusterLabels</code> instance.
+    */
+    static constructFromObject(data: any, obj: any): any;
```
Comment (axbarsan) [inline @types/model/V5ClusterLabels.d.ts:14]:
```suggestion
    static constructFromObject(data: Partial<V5ClusterLabels>, obj: V5ClusterLabels): V5ClusterLabels;
```

File: @types/model/V5ClusterLabelsResponse.d.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
+    * @param {Object} data The plain JavaScript object bearing properties of interest.
+    * @param {module:model/V5ClusterLabelsResponse} obj Optional instance to populate.
+    * @return {module:model/V5ClusterLabelsResponse} The populated <code>V5ClusterLabelsResponse</code> instance.
+    */
+    static constructFromObject(data: any, obj: any): any;
+    /**
+    * @member {module:model/V5ClusterLabels} labels
+    */
+    labels: any;
```
Comment (axbarsan) [inline @types/model/V5ClusterLabelsResponse.d.ts:18]:
```suggestion
    labels: V5ClusterLabels;
```

File: @types/model/V5ClusterLabelsResponse.d.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+*/
+export default class V5ClusterLabelsResponse {
+    /**
+    * Constructs a <code>V5ClusterLabelsResponse</code> from a plain JavaScript object, optionally creating a new instance.
+    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
+    * @param {Object} data The plain JavaScript object bearing properties of interest.
+    * @param {module:model/V5ClusterLabelsResponse} obj Optional instance to populate.
+    * @return {module:model/V5ClusterLabelsResponse} The populated <code>V5ClusterLabelsResponse</code> instance.
+    */
+    static constructFromObject(data: any, obj: any): any;
```
Comment (axbarsan) [inline @types/model/V5ClusterLabelsResponse.d.ts:14]:
```suggestion
    static constructFromObject(data: Partial<V5ClusterLabelsResponse>, obj: V5ClusterLabelsResponse): V5ClusterLabelsResponse;
```

File: @types/model/V5ListClustersByLabel.d.ts [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    * @alias module:model/V5ListClustersByLabel
+    * @class
+    * @param labels {String} The label selector string for filtering the clusters
+    */
+    constructor(labels: string);
+    /**
+    * The label selector string for filtering the clusters
+    * @member {String} labels
+    */
+    labels: any;
```
Comment (axbarsan) [inline @types/model/V5ListClustersByLabel.d.ts:26]:
```suggestion
    labels: string;
```

File: @types/model/V5ListClustersByLabel.d.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+*/
+export default class V5ListClustersByLabel {
+    /**
+    * Constructs a <code>V5ListClustersByLabel</code> from a plain JavaScript object, optionally creating a new instance.
+    * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
+    * @param {Object} data The plain JavaScript object bearing properties of interest.
+    * @param {module:model/V5ListClustersByLabel} obj Optional instance to populate.
+    * @return {module:model/V5ListClustersByLabel} The populated <code>V5ListClustersByLabel</code> instance.
+    */
+    static constructFromObject(data: any, obj: any): any;
```
Comment (axbarsan) [inline @types/model/V5ListClustersByLabel.d.ts:14]:
```suggestion
    static constructFromObject(data: Partial<V5ListClustersByLabel>, obj: V5ListClustersByLabel): V5ListClustersByLabel;
```

File: @types/api/ClustersApi.d.ts [hunk 11]
```diff
@@ -232,6 +232,36 @@ export default class ClustersApi {
...
+     * @param {module:model/V5ListClustersByLabel} body Label selector
+     * @param {Object} opts Optional parameters
+     * @param {String} opts.xRequestID A randomly generated key that can be used to track a request throughout services of Giant Swarm.
+     * @param {String} opts.xGiantSwarmActivity Name of an activity to track, like \&quot;list-clusters\&quot;. This allows to analyze several API requests sent in context and gives an idea on the purpose.
+     * @param {String} opts.xGiantSwarmCmdLine If activity has been issued by a CLI, this header can contain the command line
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link Array.<module:model/V4ClusterListItem>} and HTTP response
+     */
+    getV5ClustersByLabelWithHttpInfo(body: any, opts: {
```
Comment (axbarsan) [inline @types/api/ClustersApi.d.ts:245]:
```suggestion
    getV5ClustersByLabelWithHttpInfo(body: V5ListClustersByLabel, opts: {
```

File: @types/api/ClustersApi.d.ts [hunk 12]
```diff
@@ -232,6 +232,36 @@ export default class ClustersApi {
...
+     * @param {module:model/V5ListClustersByLabel} body Label selector
+     * @param {Object} opts Optional parameters
+     * @param {String} opts.xRequestID A randomly generated key that can be used to track a request throughout services of Giant Swarm.
+     * @param {String} opts.xGiantSwarmActivity Name of an activity to track, like \&quot;list-clusters\&quot;. This allows to analyze several API requests sent in context and gives an idea on the purpose.
+     * @param {String} opts.xGiantSwarmCmdLine If activity has been issued by a CLI, this header can contain the command line
+     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link Array.<module:model/V4ClusterListItem>}
+     */
+    getV5ClustersByLabel(body: any, opts: {
```
Comment (axbarsan) [inline @types/api/ClustersApi.d.ts:260]:
```suggestion
    getV5ClustersByLabel(body: V5ListClustersByLabel, opts: {
```

File: @types/model/V5ClusterDetailsResponse.d.ts [hunk 13]
```diff
@@ -69,4 +69,8 @@ export default class V5ClusterDetailsResponse {
     * @member {Array.<module:model/V5ClusterDetailsResponseVersions>} versions
     */
     versions: V5ClusterDetailsResponseVersions[];
+    /**
+    * @member {module:model/V5ClusterLabels} labels
+    */
+    labels: {[labelKey: string]: string};
```
Comment (ubergesundheit) [inline @types/model/V5ClusterDetailsResponse.d.ts:75]: @axbarsan this is not right, right?

Comment (axbarsan) [reply @types/model/V5ClusterDetailsResponse.d.ts:75]: but why not the `V5ClusterLabels` type?

Comment (ubergesundheit) [reply @types/model/V5ClusterDetailsResponse.d.ts:75]: Better import `V5ClusterLabels` and reference it?

Comment (axbarsan) [reply @types/model/V5ClusterDetailsResponse.d.ts:75]: yasss

**Pull request merged**